osquery/tables/system/darwin/screenlock.mm

```diff
@@ -0,0 +1,79 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#import <Foundation/Foundation.h>
+#include <limits.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <stdbool.h>
+
+namespace osquery {
+namespace tables {
+
+void logUnsupportedError() {
+  LOG(ERROR) << "This table is not supported on this version of macOS";
+  return;
+}
+
+QueryData genScreenlock(QueryContext& context) {
+  QueryData results;
+  Row r;
+
+  auto bundle_url = CFURLCreateWithFileSystemPath(
+      kCFAllocatorDefault,
+      CFSTR("/System/Library/PrivateFrameworks/MobileKeyBag.framework"),
+      kCFURLPOSIXPathStyle,
+      true);
+
+  if (bundle_url == nullptr) {
+    logUnsupportedError();
+    VLOG(1) << "Error parsing MobileKeyBag bundle URL";
+
+    return results;
+  }
+
+  auto bundle = CFBundleCreate(kCFAllocatorDefault, bundle_url);
+  CFRelease(bundle_url);
+
+  if (bundle == nullptr) {
+    logUnsupportedError();
+    VLOG(1) << "Error opening MobileKeyBag bundle";
+
+    return results;
+  }
+
+  auto MKBDeviceGetGracePeriod =
+      (NSDictionary * (*)(NSDictionary*)) CFBundleGetFunctionPointerForName(
+          bundle, CFSTR("MKBDeviceGetGracePeriod"));
+  if (MKBDeviceGetGracePeriod == nullptr) {
+    logUnsupportedError();
+    VLOG(1) << "MKBDeviceGetGracePeriod returned null";
+    CFRelease(bundle);
+
+    return results;
+  }
+
+  NSDictionary* durationDict;
+
+  // MKBDeviceGetGracePeriod requires an empty dictionary as the sole argument
+  durationDict = MKBDeviceGetGracePeriod(@{});
+  int duration = [durationDict[@"GracePeriod"] integerValue];
```

Comment:
Since this is undocumented, I would be extra paranoid about checking types to avoid crashing from unimplemented methods:

NSDictionary *durationDict = MKBDeviceGetGracePeriod(@{});
if ([durationDict isKindOfClass:[NSDictionary class]]) ... handle error

and assuming "GracePeriod" is of type NSNumber:
NSNumber *durationNumber = durationDict[@"GracePeriod"];
if ([durationNumber isKindOfClass:[NSNumber class]]) ... handle error
int duration = durationNumber.integerValue;